controllers/user.go

```diff
@@ -1,31 +1,112 @@
+		return
+	}
+
+	var users models.User
+	db.DB.Find(&users, "email=?", input.Email)
+
+	if users.Email == input.Email {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Email is Already Registered", "success": false})
+		return
+	}
+
+	bytePassword := []byte(input.Password)
+	hashedPassword, err := bcrypt.GenerateFromPassword(bytePassword, bcrypt.DefaultCost)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	// Create new user
+	user := models.User{Name: input.Name, Email: input.Email, Password: string(hashedPassword)}
+
+	db.DB.Create(&user)
+
+	c.IndentedJSON(http.StatusOK, gin.H{"success": true})
+}
+
+func (ctrl UserController) LoginUser(c *gin.Context) {
+
+	var input models.LoginInput
+	err := c.ShouldBind(&input)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	var user models.User
+	if db.DB.Find(&user, "email=?", input.Email).RecordNotFound() {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "You have not registered, please register first", "success": false})
+		return
+	}
+
+	//Compare the password form and database if match
+	bytePassword := []byte(input.Password)
+	byteHashedPassword := []byte(user.Password)
+	err = bcrypt.CompareHashAndPassword(byteHashedPassword, bytePassword)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "Incorrect Password", "success": false})
+		return
+	}
+	token, err := util.CreateToken(input.Email)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"token": token, "success": true})
+}
+
+func (ctrl UserController) GetAll(c *gin.Context) {
+	var users []models.User
+	db.DB.Find(&users)
+
+	c.IndentedJSON(http.StatusOK, gin.H{"data": users, "success": true})
 }
 
-func CreateUser(c *gin.Context) {
-  // Validate input
-  var input models.UserInput
-  err := c.ShouldBindJSON(&input)
-  if err != nil {
-    c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-    return
-  }
+func (ctrl UserController) GetOne(c *gin.Context) {
+	id := c.Param("id")
+
+	getID, err := strconv.ParseInt(id, 10, 64)
+	if getID == 0 || err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Invalid Parameter", "success": false})
+		return
+	}
+
+	var user models.User
+
+	if db.DB.Find(&user, "id=?", getID).RecordNotFound() {
+		c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"error": "No User Found", "success": false})
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, gin.H{"data": user, "success": true})
+}
 
-  // Create new user
-  user := models.User{ID: input.ID, Name: input.Name, Email: input.Email, Password: input.Password}
-  db.DB.Create(&user)
+func (ctrl UserController) LogoutUser(c *gin.Context) {
+	// Delete Authentication token
```

Comment:
The JWT is stored on browser, so remove the token deleting the cookie at client side.
Usually you delete JWT token from backend if you stored them in Redis.
Or if you really want to delete the token from backend you can create a blocklist of JWTs and check if the token is inside that list.

We can save effort and just implement if from client side and ignore this endpoint :) 🗡️

Reply:
Okay it will be handled from client side